docs: documenting the flags issue in CAA record (#384)
terraform-ibm-modules/terraform-ibm-cis: modules/dns/README.md

This module is used to create and manage IBM Cloud Internet Services (CIS) DNS records.

1. Although the SRV record name is provided in the variable, it is stored as `_service._proto.record_name.domain_name TTL class type of record priority weight port target`. For more information, see  [What is a DNS SRV record?](https://www.cloudflare.com/en-gb/learning/dns/dns-records/dns-srv-record/).

    The changed name means that when you run a `terraform plan` command after a successful `terraform apply`, the output shows that the DNS record requires an update, as shown in the following example. However, your infrastructure will not be affected.

    ```
    # module.cis_dns_records[0].ibm_cis_dns_record.dns_records["NAME/SRV"] will be updated in-place
    ~ resource "ibm_cis_dns_record" "dns_records" {
        id          = "a5177ec049fc2973a33df1441e869a27:9684838a87ecxxx5518:crn:v1:bluemix:public:internet-svcs:global:a/abac0df06b644axxx4f55b3880e:6ee7ec9a-5e68-4b6f-af9a-5714xxx4d::"
        ~ name        = "_sip._udp.test-example.srv.test**.**.com" -> "test-example.srv"
        # (13 unchanged attributes hidden)
        }
    ```

> 2. When a DNS record of CAA type is added, the `flags` is automatically assigned.

Author: I have made the changes as per the comment.

Although the SRV record name is provided in the variable, it is stored as `_service._proto.record_name.domain_name TTL class type of record priority weight port target`. For more information, see  [What is a DNS SRV record?](https://www.cloudflare.com/en-gb/learning/dns/dns-records/dns-srv-record/).

The changed name means that when you run a `terraform plan` command after a successful `terraform apply`, the output shows that the DNS record requires an update, as shown in the following example. You can ignore that message. Your infrastructure will not be affected.

    # module.cis_dns_records[0].ibm_cis_dns_record.dns_records["NAME/SRV"] will be updated in-place
    ~ resource "ibm_cis_dns_record" "dns_records" {
        id          = "a5177ec049fc2973a33df1441e869a27:9684838a87ecxxx5518:crn:v1:bluemix:public:internet-svcs:global:a/abac0df06b644axxx4f55b3880e:6ee7ec9a-5e68-4b6f-af9a-5714xxx4d::"
        ~ name      = "_sip._udp.test-example.srv.test**.**.com" -> "test-example.srv"
        # (13 unchanged attributes hidden)
        }

> If you add a CAA record, a `flags` parameter is returned in the data object. The work is being tracked on this issue [here](IBM-Cloud/terraform-provider-ibm#4792).

Reviewer: ```suggestion
If you add a CAA record, a `flags` parameter is returned in the data object. The work is being tracked in `IBM-Cloud/terraform-provider-ibm` [issue 4792] (https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4792).
```

The changed name means that when you run a `terraform plan` command after a successful `terraform apply`, the output shows that the DNS record requires an update, as shown in the following example. However, your infrastructure will not be affected.

    # module.cis_dns_records[0].ibm_cis_dns_record.dns_records["NAME/SRV"] will be updated in-place
    ~ resource "ibm_cis_dns_record" "dns_records" {
        id          = "a5177ec049fc2973a33df1441e869a27:9684838a87ecxxx5518:crn:v1:bluemix:public:internet-svcs:global:a/abac0df06b644axxx4f55b3880e:6ee7ec9a-5e68-4b6f-af9a-5714xxx4d::"
        ~ name      = "_sip._udp.test-example.srv.test**.**.com" -> "test-example.srv"
        # (13 unchanged attributes hidden)
        }

> If a record of type CAA is added, the flags parameter is augmented in the data object. More information on the flags parameter can be found [here](https://developers.cloudflare.com/api/operations/dns-records-for-a-zone-create-dns-record).

Reviewer: I had to dig in the issue to find the context with Cloudflare: 

> The field flags  seems to have not been added on CIS end but it is present in the Cloudflare request data for DNS records

> ...the value has been hardcoded to 0. This is because as per the RFC for CAA records it looks like its the only valid flag value at the moment.

I think this needs more context. What is this flag? Why does it get returned. 

You can link to the ibm provider issue (https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4792) if we want to say that worked is being tracked on an issue that will be fixed. Add that only if it helps developers though.

---

perhaps...

> If you add a CAA record, a `flags` parameter is returned in the data object. `[add context here]`. For more information about the flags parameter, see [Create DNS Record](https://developers.cloudflare.com/api/operations/dns-records-for-a-zone-create-dns-record).

---

If you can link to a Cloud docs topic that would be better. And there's nothing in the cloudflare topic about flags that I saw. 

I found this topic, but it doesn't mention flags. https://cloud.ibm.com/docs/cis?topic=cis-set-up-your-dns-for-cis#caa-type-record

Reviewer 2: Actually clarification is not there. We do not have any related IBM doc...it takes us to Cloudflare doc. 
But that is a good idea to link with the provider issue. Did that for now.

> The addition of flags parameter means that when you run a `terraform plan` command after a successful `terraform apply`, the output shows that the DNS record requires an update, as shown in the following example. However, your infrastructure will not be affected.

Reviewer: ```suggestion
The returned `flags` parameter means that when you run a `terraform plan` command after `apply`, the output shows that the DNS record requires an update, as shown in the following example. You can ignore that message. Your infrastructure will not be affected.
```

Not sure if what I have here is correct. If it is, you can update line 9 as well to be consistent.

Reviewer 2: Made the changes.
